peekingduck/pipeline/nodes/draw/legend.py

```diff
@@ -0,0 +1,55 @@
+"""
+Copyright 2021 AI Singapore
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from typing import Any, Dict, List
+from peekingduck.pipeline.nodes.node import AbstractNode
+from peekingduck.pipeline.nodes.draw.utils.legend import Legend
+
+
+class Node(AbstractNode):
+    """Draw node for drawing Legend box and info on image"""
+
+    def __init__(self, config: Dict[str, Any]) -> None:
+        super().__init__(config, node_path=__name__)
+        self.all_legend_items = config['all_legend_items']
+        self.include: List[str] = config['include']
+        self.position = config['position']
+        self.legend_items: List[str] = []
+
+    def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+        """Draws legend box with information from nodes
+
+        Args:
+            inputs (dict): Dict with all available keys
+
+        Returns:
+            outputs (dict): Dict with keys "none"
+        """
+        if len(self.legend_items) == 0:
+            # Check inputs to set legend items to draw
+            if self.include[0] == 'all':
+                self.include = self.all_legend_items
+            self._include(inputs)
+        if len(self.legend_items) != 0:
+            Legend().draw(inputs, self.legend_items, self.position)  # type: ignore
+        else:
+            return {}
+        return {'img': inputs['img']}
```

Comment:
This breaks from our convention in 2 ways:
1) It returns "img" although other draw nodes return "none"
2) It returns an empty dict in some cases

What was the reason that this node had to return "img" again? I suggest including comments for both 1) and 2) to avoid confusion down the road.

Reply:
This is good question. I'm including explanation here in for posterity:

some methods can directly replace the image from the reference. However, some methods do not, much like how in pandas, some method have `replace=True`. In array reference, some manipulation is on the object referenced, some does NOT replace the element value of the reference object.

In this case, For weighted construction from 2 images used to create the legend box, the original storage object values are not updated and thus need to hard replace the dictionary img reference link to the newly created image. As such, we need to return the new img object for pipeline to replace.

I have tried without avail to standardise this, but couldn't find a similar way such that the original image referenced is updated directly. I have left a short comment on this above this code snippet